docs/API.md

```diff
@@ -576,3 +578,31 @@ performance penalty.
 
 Returns an empty string. This function is used when the `timestamp` option
 is set to `false`.
+
+<a id="metadata"></a>
+# Metadata
+
+A destination stream can have a property `Symbol.for('needsMetadata')`
+to indicate that for every log line that is written, it wants set to
```

Comment:
```
... for every log line written, the following properties of the stream should be set:

* `stream.lastLevel`: the last logging level used
...
```

Also, what does "last" refer to here? The current level being used, or the level previous to the current level being used? That is, if we have:

```
log.debug('foo')
log.info('bar')
```

Which of those levels would be assigned to `stream.lastLevel` when `log.info('bar')` is logged? The same question applies to the other `last` properties.